tests/glsl-intrinsic/atomic/atomicCounter.slang

```diff
@@ -0,0 +1,130 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl -DTARGET_GLSL
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv -emit-spirv-directly -DTARGET_SPIRV
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=4):out,name=outputBuffer
+buffer MyBlockName
+{
+    uint data[2];
+} outputBuffer;
+
+// CHECK_GLSL-DAG: void main(
+// CHECK_SPV-DAG: OpEntryPoint
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0 0 0], stride=4):name=one
+layout(binding = 1, offset = 12) uniform atomic_uint one;
+bool testSetterAndGetter()
+{
+    return true
+
+// CHECK_GLSL-DAG: atomicExchange
+// CHECK_SPV-DAG: OpAtomicExchange
+        && atomicCounterExchange(one, 1) == 0
+
+// CHECK_GLSL-DAG: atomicExchange
+// no idea how to check the spirv reliabley...
+        && atomicCounter(one) == 1
+        && atomicCounterExchange(one, 5) == 1
+        && atomicCounter(one) == 5
+        ;
+}
+
+bool counterAsParam(atomic_uint param)
+{
+    return true
+        && atomicCounterExchange(param, 5) != 100
+
+// CHECK_GLSL-DAG: atomicAdd(
+// CHECK_SPV-DAG: OpAtomicIAdd
+        && atomicCounterIncrement(param) == 5
+        && atomicCounter(param) == 6
+        ;
+}
+
+bool testAtomicUint()
+{
+    // ensure the code emits for `one` index into [3] for 12/4
+    // CHECK_GLSL-DAG: {{.*}}_data_0[3]{{.*}}
+    return true
+    
+        && atomicCounterExchange(one, 5) != 100
+// CHECK_GLSL-DAG: atomicAdd(
+// CHECK_SPV-DAG: OpAtomicIIncrement
```

Comment:
Please add the suggested lines.

Reply:
this is no longer an issue since each function shows up only a limited number of times, if I check for add twice, and I find add twice, we know the code is working?

Reply:
I will add a label check to ensure order though